crates/db_views/src/person/person_view.rs

```diff
@@ -37,33 +65,63 @@ impl PersonView {
 
     query.first(conn).await
   }
+}
 
-  pub async fn admins(pool: &mut DbPool<'_>) -> Result<Vec<Self>, Error> {
-    let conn = &mut get_conn(pool).await?;
-    Self::joins()
-      .filter(person::deleted.eq(false))
-      .filter(local_user::admin.eq(true))
-      .order_by(person::published)
-      .select(Self::as_select())
-      .load::<Self>(conn)
-      .await
-  }
+#[derive(Default)]
+pub struct PersonQuery {
+  pub admins_only: Option<bool>,
+  pub banned_only: Option<bool>,
+  pub cursor_data: Option<Person>,
+  pub page_back: Option<bool>,
+  pub limit: Option<i64>,
+  pub ignore_page_limits: Option<bool>,
+}
 
-  pub async fn banned(pool: &mut DbPool<'_>) -> Result<Vec<Self>, Error> {
+impl PersonQuery {
+  pub async fn list(self, pool: &mut DbPool<'_>) -> Result<Vec<PersonView>, Error> {
     let conn = &mut get_conn(pool).await?;
-    Self::joins()
+
+    let mut query = PersonView::joins()
       .filter(person::deleted.eq(false))
-      .filter(
-        person::banned.eq(true).and(
+      .select(PersonView::as_select())
+      .into_boxed();
+
+    // Filters
+    if self.banned_only.unwrap_or_default() {
+      query = query.filter(
+        person::local.and(person::banned).and(
           person::ban_expires
             .is_null()
             .or(person::ban_expires.gt(now().nullable())),
         ),
-      )
-      .order_by(person::published)
-      .select(Self::as_select())
-      .load::<Self>(conn)
-      .await
+      );
+    }
+
+    if self.admins_only.unwrap_or_default() {
+      query = query.filter(local_user::admin);
+    }
+
+    if !self.ignore_page_limits.unwrap_or(false) {
```

Comment:
You can remove this param and ignore page limits whenever `admins_only == true`

Reply:
Good  call